src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -117,9 +119,9 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
             // already attached
             return;
         }
-
+        const disableNearInteraction = this._options.disableNearInteraction as boolean;

```

Comment:
Is there a need to mark this as a boolean here, since you defined the parameter as a boolean above?

Reply:
but when undefined it is not setting to a boolean value instead it is assigned to be undefined which the below method would not take in. So, i had to take its value as boolean in order to account for when it is undefined

Reply:
Ah, that's because disableNearInteraction is an optional parameter. You can usually account for that with "!!", which converts the value to a boolean (essentially the same thing you're doing here).
Typescript usually drops all the "as ..." code when generating the JS files, so I'm concerned that this approach will lose information, since I think "as ..." is basically telling it to interpret the undefined value as a boolean value (a hard cast).